### PR TITLE
fix(gsd): preserve ROADMAP.md sections after projection hook

### LINKED ISSUE #4402: plan-milestone roadmap gets clobbered by workflow projection, stripping Boundary Map
## Problem
`gsd_plan_milestone` persists `boundaryMapMarkdown`, but the final rendered `ROADMAP.md` for `.gsd/worktrees/M001-i8ewog` drops `## Boundary Map`, so S06/T01 fails and T02 remains blocked.

## Root Cause
The milestone planning tool renders the correct DB-backed roadmap first, then immediately overwrites it with a simplified projection that does not include `boundary_map_markdown`.

### Failure path
- `tools/plan-milestone.js:231-243` renders the authoritative roadmap via `renderRoadmapFromDb(basePath, params.milestoneId)` and then calls `renderAllProjections(basePath, params.milestoneId)`.
- `markdown-renderer.js:124-127` correctly renders:

```js
if (milestone.boundary_map_markdown.trim()) {
    lines.push("## Boundary Map");
    lines.push("");
    lines.push(milestone.boundary_map_markdown.trim());
}
```

- `workflow-projections.js:327-330` then overwrites `ROADMAP.md` via `renderRoadmapProjection(basePath, milestoneId)`.
- `workflow-projections.js:88-128` builds a reduced roadmap containing only title, vision, and slice overview:

```js
export function renderRoadmapContent(milestoneRow, sliceRows) {
    const lines = [];
    lines.push(`# ${milestoneRow.id}: ${displayTitle}`);
    lines.push("");
    lines.push("## Vision");
    lines.push(milestoneRow.vision || milestoneRow.title || "TBD");
    lines.push("");
    lines.push("## Slice Overview");
    // ... table only ...
    lines.push("");
    return lines.join("\n");
}
```

This projection never emits `## Boundary Map`, so the post-mutation hook clobbers the correct file.

### Related stale code path
`migrate/writer.js:87` still contains:

```js
// Skip Boundary Map section entirely per D004
```

This looks like a second divergent roadmap writer. It may not be the live overwrite path for this symptom, but it preserves the same incorrect assumption and makes debugging harder.

## Expected Behavior
After `gsd_plan_milestone` writes a roadmap via `renderRoadmapFromDb()`, no later hook should overwrite that file with a lossy projection.

Concrete fix suggestion:
1. In `workflow-projections.js`, stop rewriting `ROADMAP.md` from `renderAllProjections()`.
   - This mirrors the existing protection for `PLAN.md` at `workflow-projections.js:338-341`, where projections already avoid overwriting the authoritative markdown renderer output.
2. Alternatively, make `renderRoadmapProjection()` delegate to the authoritative roadmap renderer instead of its own reduced `renderRoadmapContent()` implementation.
3. Add a regression test that calls milestone planning with non-empty `boundaryMapMarkdown` and asserts the final on-disk `ROADMAP.md` still contains:
   - `## Boundary Map`
   - at least one `### S01 → S02`
4. Remove or update the stale Boundary Map skip in `migrate/writer.js` so all roadmap renderers agree.

## Environment
- GSD version: 2.75.0
- Model: gpt-5.4
- Unit: execute-task/M001-i8ewog/S06/T01

## Reproduction Context
Milestone `M001-i8ewog`, slice `S06`, task `T01` was restoring the roadmap Boundary Map through the canonical `gsd_plan_milestone` path. The tool succeeded, but immediate verification against the active-worktree `ROADMAP.md` failed because the rendered file no longer contained `## Boundary Map` or any of the 11 expected H3 edge sections.

## Forensic Evidence
- Activity log for `execute-task/M001-i8ewog/S06/T01` shows repeated assertion failures after roadmap planning:
  - `AssertionError`
  - `roadmap boundary map missing after gsd_plan_milestone`
- The active-worktree roadmap row could be restored, but `## Boundary Map` remained absent.
- Source confirms the overwrite sequence:
  - `tools/plan-milestone.js:231-243`
  - `workflow-projections.js:327-330`
  - `workflow-projections.js:88-128`
- Existing issue `#3216` / PR `#3281` are related to missing roadmap sections in `markdown-renderer`, but they do not cover this overwrite path or Boundary Map clobbering.

---
*Auto-generated by `/gsd forensics`*

### PR DESCRIPTION
**Pull request by NilsR0711**
## TL;DR

**What:** Stop the post-mutation projection hook from overwriting the authoritative `ROADMAP.md` and stripping `## Boundary Map`.
**Why:** `gsd_plan_milestone` persists `boundaryMapMarkdown` and renders the full roadmap, but the reduced projection clobbers the file seconds later, breaking downstream consumers like S06/T01 in M001-i8ewog.
**How:** Route `renderAllProjections` and `regenerateIfMissing` through the authoritative `renderRoadmapFromDb` instead of the reduced `renderRoadmapProjection`, mirroring the existing `PLAN.md` protection.

## What

- `src/resources/extensions/gsd/workflow-projections.ts` — `renderAllProjections` and the `regenerateIfMissing` ROADMAP branch now delegate to `renderRoadmapFromDb` from `markdown-renderer.ts`. The ROADMAP branch returns `false` (fire-and-forget, matching the STATE branch).
- `src/resources/extensions/gsd/markdown-renderer.ts` — `renderRoadmapMarkdown`'s checkbox predicate uses `isClosedStatus(slice.status)` instead of a bare `=== "complete"`, aligning with `reactive-graph`, `doctor`, `auto-dashboard`, and the status-guards convention.
- New regression test `tests/plan-milestone-boundary-map-preservation.test.ts` — runs `handlePlanMilestone` with a non-empty `boundaryMapMarkdown` and asserts the final on-disk `ROADMAP.md` still contains the `## Boundary Map` heading and its rows.
- Updated existing assertions in `plan-milestone.test.ts`, `complete-slice.test.ts`, and `projection-no-plan-overwrite.test.ts` to match the authoritative checkbox-list output (previously they asserted the reduced table output that the clobber produced).

The reduced `renderRoadmapProjection` / `renderRoadmapContent` functions stay exported for their own test coverage but are no longer on the live write path.

## Why

`gsd_plan_milestone` (and `gsd_reassess_roadmap`) call `renderRoadmapFromDb`, which writes the authoritative roadmap including the required `## Boundary Map` section documented in `GSD-WORKFLOW.md`. The post-mutation hook `renderAllProjections` then called `renderRoadmapProjection`, which writes only title + vision + a slice-overview table to the same file path. Whichever ran last won — and the projection always ran last.

Forensics on `execute-task/M001-i8ewog/S06/T01` showed the authoritative roadmap row in the DB was correct, but the on-disk `ROADMAP.md` was missing `## Boundary Map` and all 11 expected H3 edge sections, hard-blocking T02.

Closes #4402

## How

Option 2 from the issue (delegate the projection to the authoritative renderer) rather than Option 1 (drop the call entirely). Option 1 would have required every state-transition tool (complete-slice, complete-task, reopen-*, etc.) to gain its own roadmap re-render call to keep slice checkboxes in sync. Delegating inside `renderAllProjections` keeps the existing call sites working and gives us a single roadmap renderer.

The secondary `renderRoadmapMarkdown` change fixes a latent inconsistency: the projection correctly treated `done`/`skipped` slices as closed (via `isClosedStatus`), while the authoritative renderer only checked for literal `"complete"`. Without this, delegating the live path to `renderRoadmapMarkdown` would have regressed slices with legacy status values.

`migrate/writer.ts`'s stale `// Skip Boundary Map section entirely per D004` is intentionally out of scope — it runs only during one-time migration, is not the live overwrite path, and follows a different structured-entry model. A separate consistency fix is warranted per "one concern per PR".

## Test plan

- [x] New regression test fails before the fix, passes after
- [x] `npx tsc --noEmit` clean
- [x] Full GSD test suite: 5122/5123 passing (one pre-existing failure in `flat-rate-routing-guard.test.ts` for #3453, unrelated and present on `main`)
- [x] Updated existing assertions validated against the new authoritative output

## Change type

- [x] `fix` — Bug fix

## AI-assisted

This PR was developed with AI assistance. The root-cause analysis, code changes, and tests were reviewed.

## Summary by CodeRabbit

* **Bug Fixes**
  * Roadmap and plan checkboxes now treat all "closed" statuses consistently and stale-detection messages reference "closed".
  * Projection regeneration now awaits rendering and accurately reports whether files were produced, improving reliability.

* **Style**
  * Roadmap output moved from table/emoji rows to markdown checkbox-style lists with inline metadata.

* **Tests**
  * Added and updated tests covering boundary-map preservation, checklist rendering, projection/regeneration behavior, and end-to-end roadmap assertions.

### CODE REVIEW

File: src/resources/extensions/gsd/workflow-projections.ts [hunk 1]
```diff
@@ -416,15 +419,16 @@ export async function renderAllProjections(basePath: string, milestoneId: string
 
 /**
  * Check if a projection file exists on disk. If missing, regenerate it from DB.
- * Returns true if the file was regenerated, false if it already existed.
+ * Returns true if the file was regenerated, false if it already existed or
+ * regeneration failed.
```
[the remark this answers is not in the document]
Comment (NilsR0711) [reply src/resources/extensions/gsd/workflow-projections.ts:423]: Thanks for the catch — substantively correct: `renderStateProjection` has four silent-return paths (`!isDbAvailable()`, missing adapter, DB probe failure, outer `catch`) that resolve without writing `STATE.md`, so `await renderStateProjection(...); return true;` would be a false success. `renderPlanProjection` also has one silent no-op (`if (!sliceRow) return;`); `renderRoadmapFromDb` throws on failure and is caught by the outer `try/catch`, so the `existsSync` check is defensive there but consistent.

This was already addressed in [8d9938c77](https://github.com/gsd-build/gsd-2/pull/4449/commits/8d9938c77) — the switch now reads:

```ts
case "PLAN":    renderPlanProjection(basePath, milestoneId, sliceId); return existsSync(filePath);
case "ROADMAP": await renderRoadmapFromDb(basePath, milestoneId);     return existsSync(filePath);
case "STATE":   await renderStateProjection(basePath);                return existsSync(filePath);
```

which matches the suggested diff exactly.
